asm_to_pil/src/vm_to_constrained.rs

```diff
@@ -567,7 +566,6 @@ impl<T: FieldElement> ASMPILConverter<T> {
             Expression::Reference(reference) => {
                 assert!(reference.namespace().is_none());
                 assert!(reference.index().is_none());
-                assert!(!reference.shift());
```

Comment:
I wonder if the two assertions here would be caught earlier

Reply:
Catching assertions earlier through actual type checking is planned as a later step, I just want to make sure it works in general.